cdc/sink/codec/canal_flat.go

```diff
@@ -258,8 +349,13 @@ func (c *CanalFlatEventBatchEncoder) Reset() {
 	panic("not supported")
 }
 
-// SetParams is no-op for now
 func (c *CanalFlatEventBatchEncoder) SetParams(params map[string]string) error {
-	// no op
+	if s, ok := params["enable-tidb-extension"]; ok {
+		a, err := strconv.ParseBool(s)
+		if err != nil {
+			return err
```

Comment:
Maybe we can wrap this error so that an RFC code can be provided for the user to identify the problem?